dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,518 @@
+                        .andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void testDelete()
+            throws Exception {
+        System.out.println("delete");
+
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .build();
+        getClient().perform(delete(URI_ROOT + '/' + request.getToken()))
+                .andExpect(status().isMethodNotAllowed());
+    }
+
+    /**
+     * Verify that Spring Security's CORS settings are working as we expect.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testUntrustedOrigin()
+            throws Exception {
+        System.out.println("testUntrustedOrigin");
+
+        // First, get a valid login token
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        // Verify token works
+        getClient(token).perform(get("/api/authn/status"))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$.okay", is(true)))
+                        .andExpect(jsonPath("$.authenticated", is(true)))
+                        .andExpect(jsonPath("$.type", is("status")));
+
+        // Test token cannot be used from an *untrusted* Origin
+        // (NOTE: this Origin is NOT listed in our 'rest.cors.allowed-origins' configuration)
+        getClient(token).perform(get(URI_ROOT)
+                .header("Origin", "https://example.org"))
+                // should result in a 403 error as Spring Security
+                //returns that for untrusted origins
+                .andExpect(status().isForbidden());
+
+        //Logout
+        getClient(token).perform(post("/api/authn/logout"))
+                        .andExpect(status().isNoContent());
+    }
+
+    /**
+     * Test of put method, of class RequestItemRepository.
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testPut()
+            throws Exception {
+        System.out.println("put");
+
+        // Create an item request to approve.
+        RequestItem itemRequest = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .build();
+
+        // Create the HTTP request body.
+        Map<String, String> parameters = Map.of(
+                "acceptRequest", "true",
+                "subject", "subject",
+                "responseMessage", "Request accepted");
+        String content = new ObjectMapper()
+                .writer()
+                .writeValueAsString(parameters);
+
+        // Send the request.
+        String authToken = getAuthToken(eperson.getEmail(), password);
+        getClient(authToken).perform(put(URI_ROOT + '/' + itemRequest.getToken())
+                .contentType(contentType)
+                .content(content))
+                .andExpect(status().isOk()
+                // .andMore?
```

Comment:
If it's possible to do so, we may want to verify that the PUT resulted in the updates we expected...not just that it returned 200.  So, is there any way to verify that the ItemRequest was approved at the Java API layer, or via another `GET` call after this `PUT`?

Also, we should add a few more `testPut()` methods here which verify incorrect behavior throws the expected errors.  For instance, we should verify that an anonymous user, or different authenticated user cannot "approve" the request.

Reply:
Approval can be tested at the API layer.  I've added that, and tests of bad PUTs.